book.md

```diff
@@ -109,7 +109,73 @@ feedback to be received through <https://github.com/llsoftsec/llsoftsecbook>.
  
 # Memory vulnerability based attacks and mitigations
 
-\missingcontent{Write chapter on memory vulnerabilities and mitigation.}
+## Introduction
+
+An important class of vulnerabilities arise due to memory access errors, such as
+buffer overflows, use-after-free accesses and null pointer dereferences. An
+attacker can exploit such vulnerabilities to leak sensitive data or
+overwrite critical memory locations and gain control of the vulnerable program.
+These types of vulnerabilities most commonly occur due to programming errors when
+using languages such as C/C++, which do not provide mechanisms to protect the
+integrity of memory accesses by default.
+
+Memory vulnerabilities have a long history. The Morris worm in 1988 was the
+first widely publicized attack exploiting a buffer overflow. Later, in the
+mid-90s a few famous write-ups describing buffer overflows appeared
+[@AlephOne1996].  Stack overflows were mitigated with stack canaries and
+non-executable stacks, but attackers answered with more ingenious ways to
+bypass these mitigations: code reuse attacks, starting from simple attacks like
+return-into-libc [@Solar1997] and leading up to Return-Oriented Programming
+[@Shacham2007] and even more complex techniques.
```

Comment:
In this sentence, a lot of new terms get introduced that will be explained in more detail later.
Would it be possible to somehow make it clear to the reader that it's OK to not understand what these terms mean here and that they will be explained later?
Maybe that could be done by making the terms links to the sections that explain them in more detail? (I'm not sure if that would work well?).
I think the following terms introduced here, and maybe in the next few sentences, could use that: stack overflows, stack canaries, non-executable stacks, code reuse attacks, return-to-libc, Return-Oriented Programming, ASLR, CFI. Maybe also "code reuse attacks"?